ct-app/core/api/protocol.py

```
@@ -0,0 +1,45 @@
+class Protocol(Enum):
+    TCP = TCPProtocol()
+    UDP = UDPProtocol()
+
+    @classmethod
+    def fromString(cls, protocol: str):
+        try:
+            return getattr(cls, protocol.upper())
+        except AttributeError:
+            raise ValueError(
+                f"Invalid protocol: {protocol}. Valid values are: {[p.name for p in cls]}"
+            )
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)